geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/tables/AttributeTable.scala

```diff
@@ -42,37 +43,33 @@ import scala.util.{Failure, Success, Try}
 object AttributeTable extends GeoMesaTable with Logging {
 
   /** Creates a function to write a feature to the attribute index **/
-  def attrWriter(bw: BatchWriter,
-                 sft: SimpleFeatureType,
-                 indexedAttributes: Seq[AttributeDescriptor],
-                 rowIdPrefix: String): FeatureWriterFn = {
-
-    val indexesOfIndexedAttributes = indexedAttributes.map { a => sft.indexOf(a.getName) }
-    val attributesToIdx = indexedAttributes.zip(indexesOfIndexedAttributes)
-
-    (toWrite: FeatureToWrite) => {
-      val mutations = getAttributeIndexMutations(toWrite, attributesToIdx, rowIdPrefix)
-      if (!mutations.isEmpty) {
-        bw.addMutations(mutations)
-      }
-    }
+  def attributeWriter(sft: SimpleFeatureType): Option[FeatureToMutations] = {
+
+    val indexedAttributes = SimpleFeatureTypes.getSecondaryIndexedAttributes(sft)
+    if (indexedAttributes.isEmpty) {
+      None
+    } else {
+      val rowIdPrefix = org.locationtech.geomesa.core.index.getTableSharingPrefix(sft)
+
+      val indexesOfIndexedAttributes = indexedAttributes.map { a => sft.indexOf(a.getName) }
+      val attributesToIdx = indexedAttributes.zip(indexesOfIndexedAttributes)
```

Comment:
You mean the flatMap in getAttributeIndexMutations? That could maybe be optimized, although in general i think map/flatMap is not *that* expensive relative to a while loop.
The snippet of code right here only gets run once per feature writer (not per feature), so optimizing this bit would not be worthwhile.